core/src/main/java/org/aerogear/mobile/core/MobileCore.java

```diff
@@ -0,0 +1,103 @@
+package org.aerogear.mobile.core;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.util.JsonReader;
+import android.util.Log;
+
+import org.aerogear.mobile.core.configuration.MobileCoreJsonParser;
+import org.aerogear.mobile.core.configuration.ServiceConfiguration;
+import org.aerogear.mobile.core.logging.Logger;
+import org.json.JSONException;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.Map;
+
+/**
+ * MobileCore is the entry point into AeroGear mobile services that are managed by the mobile-core
+ * ¿feature( TODO: Get correct noun )? in OpenShift.
+ * <p>
+ * Usage.java
+ * ```
+ * MobileCore core = new MobileCore.Builder(context, R.raw.mobile_core).build();
+ * core.feature(Keycloak.class).login();// would begin a OAuth flow.
+ * core.feature(MyRestService.class).upload(myData);// Would upload myData and be secured by KeyCloak
+ * ```
+ */
+public final class MobileCore implements ServiceModule {
+
+
+    private final Context context;
+
+    private Map<String, ServiceConfiguration> configurationMap;
+
+    private MobileCore(@NonNull Context conext) {
+        this.context = conext.getApplicationContext();
+    }
+
+    public Logger defaultLog() {
+        return (String message, Exception e) -> {
+            Log.e("MOBILE_CORE", message, e);
+        };
+    }
+
+    @Override
+    public void bootstrap(Object... args) {
+
+
+        try (InputStream configStream = context.getAssets().open("mobile-core.json");) {
```

Comment:
make the path to the JSON file configurable. Use `mobile-core.json` the default value.

Reply:
mobile-services.json as default name will make more sense. It contains info about services.